### PR TITLE
[tests-only][full-ci] test: Check absence of application/octet-stream MIME Type in app/list

### PR DESCRIPTION
**Pull request by anon-pradip**
## Description

This pull request adds test cases to ensure that the MIME type application/octet-stream is absent in the app/list endpoint. 
## Related Issue

- Fixes: https://github.com/owncloud/ocis/issues/10179
- Test coverage for: https://github.com/owncloud/ocis/issues/10086

## How Has This Been Tested?

- Locally
- CI

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: tests/acceptance/features/apiCollaboration/checkFileInfo.feature [hunk 1]
```diff
@@ -1855,3 +1855,10 @@ Feature: check file info with different wopi apps
       | Collabora     |
       | FakeOffice    |
       | OnlyOffice    |
+
+  @issue-10086
+  Scenario: check that /app/list endpoint doesn't contain the entry for application/octet-stream MIME Type
+    When user "Alice" sends HTTP method "GET" to URL "/app/list"
+    Then the HTTP status code should be "200"
+    And the response should not contain following MIME Types:
```
[the remark this answers is not in the document]
Comment (anon-pradip) [reply tests/acceptance/features/apiCollaboration/checkFileInfo.feature]: Okeii. Got it.

File: tests/acceptance/bootstrap/CollaborationContext.php [hunk 2]
```diff
@@ -323,4 +324,34 @@ public function userGetsTheInformationOfTheLastOpenedFileUsingWopiEndpoint(strin
 			)
 		);
 	}
+
+	/**
+	 * @Then /^the response should (not|)\s?contain following MIME Type:$/
```
Comment (saw-jan) [inline tests/acceptance/bootstrap/CollaborationContext.php:329]:
```suggestion
	 * @Then /^the response (should|should not) contain following MIME Type:$/
```

File: tests/acceptance/bootstrap/CollaborationContext.php [hunk 3]
```diff
@@ -323,4 +324,34 @@ public function userGetsTheInformationOfTheLastOpenedFileUsingWopiEndpoint(strin
...
+	 * @param TableNode $table
+	 *
+	 * @return void
+	 * @throws Exception
+	 */
+	public function theFollowingMimeTypesShouldExistForUser(string $shouldOrNot, TableNode $table):void {
+		$rows = $table->getRows();
+		foreach ($rows as $row) {
+			$responseArray = $this->featureContext->getJsonDecodedResponse($this->featureContext->getResponse())['mime-types'];
+			$mimeTypes = \array_column($responseArray, 'mime_type');
```
Comment (saw-jan) [inline tests/acceptance/bootstrap/CollaborationContext.php:341]: these can be done once
```suggestion
		$rows = $table->getRows();
			$responseArray = $this->featureContext->getJsonDecodedResponse($this->featureContext->getResponse())['mime-types'];
			$mimeTypes = \array_column($responseArray, 'mime_type');
		foreach ($rows as $row) {
```

File: tests/acceptance/bootstrap/CollaborationContext.php [hunk 4]
```diff
@@ -323,4 +324,34 @@ public function userGetsTheInformationOfTheLastOpenedFileUsingWopiEndpoint(strin
...
+			if ($shouldOrNot === "not") {
+				Assert::assertFalse(
+					\in_array($row[0], $mimeTypes),
+					"the response should not contain the tag $row[0].\nResponse\n"
```
Comment (saw-jan) [inline tests/acceptance/bootstrap/CollaborationContext.php:345]:
```suggestion
					"the response should not contain the mimetype '$row[0]'.\nResponse\n"
```

File: tests/acceptance/bootstrap/CollaborationContext.php [hunk 5]
```diff
@@ -323,4 +324,34 @@ public function userGetsTheInformationOfTheLastOpenedFileUsingWopiEndpoint(strin
...
+					. print_r($responseArray, true)
+				);
+			} else {
+				Assert::assertTrue(
+					\in_array($row[0], $responseArray),
+					"the response does not contain the tag $row[0].\nResponse\n"
```
Comment (saw-jan) [inline tests/acceptance/bootstrap/CollaborationContext.php:351]:
```suggestion
					"the response does not contain the mimetype $row[0].\nResponse\n"
```

File: tests/acceptance/bootstrap/CollaborationContext.php [hunk 6]
```diff
@@ -323,4 +324,34 @@ public function userGetsTheInformationOfTheLastOpenedFileUsingWopiEndpoint(strin
...
+	public function theFollowingMimeTypesShouldExistForUser(string $shouldOrNot, TableNode $table):void {
+		$rows = $table->getRows();
+		foreach ($rows as $row) {
+			$responseArray = $this->featureContext->getJsonDecodedResponse($this->featureContext->getResponse())['mime-types'];
+			$mimeTypes = \array_column($responseArray, 'mime_type');
+			if ($shouldOrNot === "not") {
+				Assert::assertFalse(
+					\in_array($row[0], $mimeTypes),
+					"the response should not contain the tag $row[0].\nResponse\n"
+					. print_r($responseArray, true)
```
Comment (saw-jan) [inline tests/acceptance/bootstrap/CollaborationContext.php:346]: log the mimetypes not the whole response
```suggestion
					. $mimeTypes
```

File: tests/acceptance/features/apiCollaboration/checkFileInfo.feature [hunk 7]
```diff
@@ -1855,3 +1855,10 @@ Feature: check file info with different wopi apps
       | Collabora     |
       | FakeOffice    |
       | OnlyOffice    |
+
+  @issue-10086
+  Scenario: check that /app/list endpoint doesn't contain the entry for application/octet-stream MIME Type
+    When user "Alice" sends HTTP method "GET" to URL "/app/list"
+    Then the HTTP status code should be "200"
+    And the response should not contain following MIME Type:
```
Comment (saw-jan) [inline tests/acceptance/features/apiCollaboration/checkFileInfo.feature:1863]:
```suggestion
    And the response should not contain the following MIME types:
```

**Review by amrita-shrestha — APPROVED** [review]

**Review by saw-jan — APPROVED** [review]

**Pull request merged**